### DISCUSSION TITLE
compilation problem

### DISCUSSION THREAD

**Discussion by goq1990** [Edzelf/ESP32Radio-V2#147] [Q&A]
what i do wrong? i have esp32-wroom-32d pcm5102a st7735:

Arduino:1.8.19 (Windows 10), Płytka:"ESP32 Dev Module, Disabled, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, Core 1, Core 1, Info, Disabled, Disabled"

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\aac_decoder.cpp: In function 'void GenerateNoiseVector(int*, int*, int)':

aac_decoder.cpp:5297:41: error: invalid conversion from 'uint32_t*' {aka 'long unsigned int*'} to 'unsigned int*' [-fpermissive]

 5297 |         coef[i] = ((int32_t)Get32BitVal((uint32_t *)last)) >> 16;

      |                                         ^~~~~~~~~~~~~~~~

      |                                         |

      |                                         uint32_t* {aka long unsigned int*}

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\aac_decoder.cpp:5130:40: note:   initializing argument 1 of 'unsigned int Get32BitVal(unsigned int*)'

 5130 | unsigned int Get32BitVal(unsigned int *last)

      |                          ~~~~~~~~~~~~~~^~~~

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\aac_decoder.cpp: In function 'int DecodeOneSymbol(int)':

aac_decoder.cpp:8533:57: error: invalid conversion from 'int*' to 'int32_t*' {aka 'long int*'} [-fpermissive]

 8533 |     nBits = DecodeHuffmanScalar(huffTabSBR, hi, bitBuf, &val);

      |                                                         ^~~~

      |                                                         |

      |                                                         int*

In file included from D:\Program Files (x86)\Arduino1\examples\ESP32Radio\aac_decoder.cpp:9:

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\aac_decoder.h:487:115: note:   initializing argument 4 of 'int DecodeHuffmanScalar(const short int*, const HuffInfo_t*, unsigned int, int32_t*)'

  487 | int DecodeHuffmanScalar(const signed short *huffTab, const HuffInfo_t *huffTabInfo, unsigned int bitBuf, int32_t *val);

      |                                                                                                          ~~~~~~~~~^~~

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\ESP32Radio.ino: In function 'void fillkeylist()':

ESP32Radio:2334:23: error: too few arguments to function 'esp_err_t nvs_entry_find(const char*, const char*, nvs_type_t, nvs_opaque_iterator_t**)'

 2334 |   it = nvs_entry_find ( "nvs", NAME, NVS_TYPE_ANY ) ;           // Get first entry

      |        ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In file included from D:\Program Files (x86)\Arduino1\examples\ESP32Radio\ESP32Radio.ino:120:

C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-442a798083\esp32/include/nvs_flash/include/nvs.h:659:11: note: declared here

  659 | esp_err_t nvs_entry_find(const char *part_name,

      |           ^~~~~~~~~~~~~~

ESP32Radio:2348:27: error: cannot convert 'nvs_iterator_t' {aka 'nvs_opaque_iterator_t*'} to 'nvs_opaque_iterator_t**'

 2348 |     it = nvs_entry_next ( it ) ;

      |                           ^~

      |                           |

      |                           nvs_iterator_t {aka nvs_opaque_iterator_t*}

C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-442a798083\esp32/include/nvs_flash/include/nvs.h:679:42: note:   initializing argument 1 of 'esp_err_t nvs_entry_next(nvs_opaque_iterator_t**)'

  679 | esp_err_t nvs_entry_next(nvs_iterator_t *iterator);

      |                          ~~~~~~~~~~~~~~~~^~~~~~~~

D:\Program Files (x86)\Arduino1\examples\ESP32Radio\ESP32Radio.ino: In function 'void setup()':

ESP32Radio:2589:32: error: 'TCPIP_ADAPTER_IF_STA' was not declared in this scope

 2589 |   tcpip_adapter_set_hostname ( TCPIP_ADAPTER_IF_STA,

      |                                ^~~~~~~~~~~~~~~~~~~~

ESP32Radio:2589:3: error: 'tcpip_adapter_set_hostname' was not declared in this scope

 2589 |   tcpip_adapter_set_hostname ( TCPIP_ADAPTER_IF_STA,

      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~

ESP32Radio:2643:22: error: too many arguments to function 'hw_timer_t* timerBegin(uint32_t)'

 2643 |   timer = timerBegin ( 0, 80, true ) ;                   // User 1st timer with prescaler 80

      |           ~~~~~~~~~~~^~~~~~~~~~~~~~~

In file included from C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.1\cores\esp32/esp32-hal.h:84,

                 from C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.1\cores\esp32/Arduino.h:36,

                 from D:\Program Files (x86)\Arduino1\examples\ESP32Radio\ESP32Radio.ino:114:

C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.1\cores\esp32/esp32-hal-timer.h:35:13: note: declared here

   35 | hw_timer_t *timerBegin(uint32_t frequency);

      |             ^~~~~~~~~~

ESP32Radio:2644:24: error: too many arguments to function 'void timerAttachInterrupt(hw_timer_t*, void (*)())'

 2644 |   timerAttachInterrupt ( timer, &timer100, false ) ;     // Call timer100() on timer alarm

      |   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~

C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.1\cores\esp32/esp32-hal-timer.h:50:6: note: declared here

   50 | void timerAttachInterrupt(hw_timer_t *timer, void (*userFunc)(void));

      |      ^~~~~~~~~~~~~~~~~~~~

ESP32Radio:2645:3: error: 'timerAlarmWrite' was not declared in this scope; did you mean 'timerWrite'?

 2645 |   timerAlarmWrite ( timer, 100000, true ) ;              // Alarm every 100 msec

      |   ^~~~~~~~~~~~~~~

      |   timerWrite

ESP32Radio:2646:3: error: 'timerAlarmEnable' was not declared in this scope; did you mean 'timerAlarm'?

 2646 |   timerAlarmEnable ( timer ) ;                           // Enable the timer

      |   ^~~~~~~~~~~~~~~~

      |   timerAlarm

ESP32Radio:2688:23: error: 'ADC_WIDTH_12Bit' was not declared in this scope; did you mean 'ADC_WIDTH_MAX'?

 2688 |   adc1_config_width ( ADC_WIDTH_12Bit ) ;

      |                       ^~~~~~~~~~~~~~~

      |                       ADC_WIDTH_MAX

Znaleziono wiele bibliotek w "WiFi.h"

Wykorzystane: C:\Users\goq19\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.1\libraries\WiFi

Niewykorzystane: D:\Program Files (x86)\Arduino1\libraries\WiFi

exit status 1

invalid conversion from 'uint32_t*' {aka 'long unsigned int*'} to 'unsigned int*' [-fpermissive]

**Comment by dvdes90** [comment]
I'm the same. I'm starting with Arduino and more with ESP32 but I'm having the same problems as you. I'm trying with visual studio +platformIO

**Reply by goq1990** [reply]
Platformio helps but now i have new problem sound no work :( only silence i have good config check mamy times
